src/python/zquantum/vqe/singlet_uccsd.py

```diff
@@ -0,0 +1,131 @@
+
+        Args:
+            n_spatial_orbitals: number of spatial orbitals.
+            n_alpha: number of alpha electrons.
+            n_layers: number of layers of the ansatz.
+            transformation: transformation used for translation between fermions and qubits.
+
+        Attributes:
+            n_beta: number of beta electrons (equal to n_alpha).
+            n_electrons: total number of electrons (n_alpha + n_beta).
+            n_qubits: number of qubits required for the ansatz circuit.
+            number_of_params: number of the parameters that need to be set for the ansatz circuit.
+        """
+        super().__init__(n_layers)
+        self._n_spatial_orbitals = n_spatial_orbitals
+        self._n_alpha = n_alpha
+        self._transformation = transformation
+        self._assert_n_spatial_orbitals()
+
+    @property
+    def n_spatial_orbitals(self):
+        return self._n_spatial_orbitals
+
+    @invalidates_parametrized_circuit
+    @n_spatial_orbitals.setter
+    def n_spatial_orbitals(self, new_n_spatial_orbitals):
+        self._n_spatial_orbitals = new_n_spatial_orbitals
+        self._assert_n_spatial_orbitals()
+
+    @property
+    def n_qubits(self):
+        return self._n_spatial_orbitals * 2
+
+    @property
+    def n_alpha(self):
+        return self._n_alpha
+
+    @invalidates_parametrized_circuit
+    @n_alpha.setter
+    def n_alpha(self, new_n_alpha):
+        self._n_alpha = new_n_alpha
+        self._assert_n_spatial_orbitals()
+
+    @property
+    def _n_beta(self):
+        return self._n_alpha
+
+    @property
+    def n_electrons(self):
+        return self._n_alpha + self._n_beta
+
+    @property
+    def number_of_params(self) -> int:
+        """
+        Returns number of parameters in the ansatz.
+        """
+        return uccsd_singlet_paramsize(
+            n_qubits=self.n_qubits, n_electrons=self.n_electrons
+        )
+
+    @overrides
+    def _generate_circuit(self, params: Optional[np.ndarray] = None) -> Circuit:
+        """
+        Returns a parametrizable circuit represention of the ansatz.
+        Args:
+            params: parameters of the circuit. 
+        """
+        circuit = Circuit()
+
+        circuit += create_layer_of_gates(self.n_qubits, "X")
+
+        # Build UCCSD generator
+        fermion_generator = uccsd_singlet_generator(
+            params, self.n_qubits, self.n_electrons, anti_hermitian=True,
+        )
+
+        evolution_operator = exponentiate_fermion_operator(
+            fermion_generator, self._transformation
+        )
+
+        circuit += Circuit(evolution_operator)
+        return circuit
+
+    def _assert_n_spatial_orbitals(self):
+        if self._n_spatial_orbitals < 2:
+            raise (
+                ValueError(
+                    "Number of spatials orbitals must be greater or equal and is {0}.".format(
+                        self._n_spatial_orbitals
+                    )
```

Comment:
Greater or equal to what?

Reply:
Adressed in [75170d8](https://github.com/zapatacomputing/z-quantum-vqe/pull/5/commits/75170d8c5c38ec0adc9f66321ca231a9fec080cf).